src/components/ConfigEditor.tsx

```diff
@@ -1,45 +1,59 @@
 import React, { ChangeEvent, PureComponent } from 'react';
 import { Icon, LegacyForms } from '@grafana/ui';
 import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
-import { LightstepDataSourceOptions, LightstepSecureJsonData } from './types';
+import { LightstepDataSourceOptions, LightstepSecureJsonData } from '../types';
 
 const { FormField, SecretFormField } = LegacyForms;
 
-interface Props extends DataSourcePluginOptionsEditorProps<LightstepDataSourceOptions> {}
+interface Props extends DataSourcePluginOptionsEditorProps<LightstepDataSourceOptions, LightstepSecureJsonData> {}
```

Comment:
Generic also accepts the secure json data type which removes the need for a type assertion in render below.